dkist/net/attrs.py

```diff
@@ -278,3 +279,60 @@ def __init__(self, bottom_left, *, top_right=None, width: u.deg = None,
 
     def collides(self, other):
         return isinstance(other, self.__class__)
+
+# averageDatasetSpectralSamplingMin, averageDatasetSpectralSamplingMax
+class SpectralSampling(_Range):
+    """
+    The average spectral sampling of a dataset.
+
+    Parameters
+    ----------
+    spectralmin : `u.Quantity`
+        The minimum value of the average spectral sampling to search between.
+
+    spectralmax : `u.Quantity`
+        The maximum value of the average spectral sampling to search between.
+    """
+    def __init__(self, spectralmin: u.nm, spectralmax: u.nm):
```

Comment:
I *think* that we want to put [`u.quantity_input`](https://docs.astropy.org/en/latest/api/astropy.units.quantity_input.html?highlight=quantity_input) on these methods, and in this case enable the spectral equivalency:
```suggestion
   u.quantity_input(equivalencies=u.spectral())
    def __init__(self, spectralmin: u.nm, spectralmax: u.nm):
```

This should validate the units against the type annotations, we should add tests for this obviously.